src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private T[] elements;
+    private int size;
+    private static final int INITIAL_CAPACITY = 10;
+
+    @SuppressWarnings("unchecked")
+    public MateStack() {
+        elements = (T[]) new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
 
+    public void push(T value) {
+        ensureCapacity();
+        elements[size++] = value;
     }
 
     public T peek() {
-        return null;
+        ensureNotEmpty();
+        return elements[size - 1];
     }
 
     public T pop() {
-        return null;
+        ensureNotEmpty();
+        T value = elements[--size];
+        elements[size] = null;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void resize() {
+        T[] newArray = (T[]) new Object[elements.length * 2];
+        System.arraycopy(elements, 0, newArray, 0, elements.length);
+        elements = newArray;
+    }
+
+    private void ensureNotEmpty() {
+        if (size == 0) {
+            throw new EmptyStackException(); // Rzucamy wyjątek
+        }
+    }
```

Comment:
The `ensureNotEmpty()` method is throwing an `EmptyStackException` without any message. It's a good practice to provide some meaningful message with the exception, which can help to debug the issue if this exception occurs. For example, you can throw the exception like this: `throw new EmptyStackException("Stack is empty");`.